src/sssom/rdf_internal.py

```diff
@@ -0,0 +1,992 @@
+        """
+        if schema.get_class(range_name) is not None:
+            # This range is for objects, not scalar values
+            return None
+
+        if range_name.endswith("_enum"):
+            return EnumValueConverter(schema, range_name)
+
+        ctor = self.constructors.get(range_name)
+        if ctor == EntityReferenceValueConverter:
+            # CURIE provider needed
+            return EntityReferenceValueConverter(ccp)
+        elif ctor is not None:
+            return ctor()
+        else:
+            # This should only happen if a brand new type of slot has
+            # been introduced in the SSSOM schema
+            raise NotImplementedError(f"Range {range_name} is not supported")
+
+
+class ObjectConverter(object):
+    """Base class for conversion of SSSOM objects to and from RDF.
+
+    One instance of this class will handle the (de)serialisation of one
+    particular type of object (e.g. MappingSet, Mapping).
+
+    The base class implements the logic that is common to all types of
+    objects. It should be subclassed for objects that require specific
+    logic.
+    """
+
+    name: str
+    """The name of the SSSOM class whose conversion is handled by this
+    object."""
+
+    object_uri: URIRef
+    """The URI representing the SSSOM class in RDF context."""
+
+    slots_by_name: Dict[str, SlotDefinition]
+    """All the valid slots for the class, addressed by their names."""
+
+    slots_by_uri: Dict[URIRef, SlotDefinition]
+    """All the valid slots for the class, addressed by the URIs that
+    represent them in RDF context."""
+
+    value_converters: Dict[str, ValueConverter]
+    """All the value converters used to convert slot values to/from RDF,
+    addressed by slot range."""
+
+    schema: SSSOMSchemaView
+    """Helper object to access information from the SSSOM schema."""
+
+    ccp: CurieConverterProvider
+    """The object to call when CURIE expansion/contraction is needed."""
+
+    def __init__(self, class_name: str, ccp: CurieConverterProvider):
+        """Create a new instance for a class of objects.
+
+        :param class_name: The name of the SSSOM class of objects to
+            convert.
+        :param ccp: A callable object that shall provide a CURIE
+            converter on demand.
+        """
+        self.schema = SSSOMSchemaView()
+        self.ccp = ccp
+
+        # Prepare the slot tables...
+        self.slots_by_name = {}
+        self.slots_by_uri = {}
+        ranges: List[SlotDefinition] = []
+        for slot in self.schema.view.class_induced_slots(class_name):
+            self.slots_by_name[slot.name] = slot
+            self.slots_by_uri[self._get_slot_uri(slot)] = slot
+            ranges.append(slot.range)
+
+        # ... and the scalar converters table
+        self.value_converters = {}
+        factory = ValueConverterFactory()
+        for rng in set(ranges):
+            vc = factory.create(rng, self.schema.view, ccp)
```

Comment:
idk why the type checker isn't catching this, but `rng` is not a string as the create() function is expecting